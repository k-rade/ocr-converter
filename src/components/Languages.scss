.option-active {
  font-weight: 600;
  color: var(--color-primary-500);
  background-color: var(--color-gray-500);
}

.languages-select {
  position: relative;
  cursor: pointer;

  .selected-language {
    border-bottom: 2px solid var(--color-gray-600);
    width: 14rem;
    padding: 0 0 0.5rem;
    border-radius: 2px;
    display: flex;
    align-items: center;
    justify-content: space-between;
    color: var(--color-white);
    font-size: 0.875rem;

    .caret-down {
      width: 12px;
      height: 6px;
      clip-path: polygon(50% 100%, 0 0, 100% 0);
      background-color: var(--color-gray-300);
      position: relative;
    }
  }

  ul {
    top: 3rem;
    position: absolute;
    list-style: none;
    border: 2px solid var(--color-gray-600);
    background-color: var(--color-gray-300);
    border-radius: 2px;
    overflow-y: auto;
    margin: 0;
    padding: 0;
    width: 14rem;
    height: 13rem;
    z-index: 1;
    box-shadow: 0 2px 6px var(--color-black-three);
    font-size: 0.875rem;
  }

  li {
    padding: 0.625rem 1rem;
  }
}

@media screen and (max-width:512px) {
  .languages {
    span {
      display: none;
    }
  }
}

@media screen and (max-width:414px) {

  .languages,
  .languages-select,
  .selected-language,
  ul {
    min-width: 100%;
  }
}