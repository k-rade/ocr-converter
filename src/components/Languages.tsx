import { useEffect, useRef, useState } from "react";
import { Language } from "../types/Language";
import { ReactComponent as ChevronDownIcon } from "../icons/chevron-down.svg";
import { LANGUAGES } from "../constants/languages";

import "./Languages.scss";

interface Props {
  setLanguage: (language: Language) => void;
  selectedLanguage: Language;
}

const Languages = ({ setLanguage, selectedLanguage }: Props): JSX.Element => {
  const [showDropdown, setShowDropdown] = useState(false);
  const ref = useRef<any>();

  const handleDropdownToggle = (): void => {
    setShowDropdown((prevState: boolean) => !prevState);
  };

  const handleLanguageSelect = (language: Language): void => {
    setLanguage(language);
    setShowDropdown(false);
  };

  useEffect(() => {
    const handleClickOutside = (event: any) => {
      if (ref.current && !ref.current.contains(event.target)) {
        setShowDropdown(false);
      }
    };
    document.addEventListener("mousedown", handleClickOutside);
    return () => document.removeEventListener("mousedown", handleClickOutside);
  }, []);

  return (
    <div className="languages">
      <span>Language:</span>
      <div className="languages-select" ref={ref}>
        <div className="selected-language" onClick={handleDropdownToggle}>
          {selectedLanguage.value}
          <div className="caret-down"></div>
        </div>
        {showDropdown && (
          <ul data-testid="languages-list">
            {LANGUAGES.map((language: Language, i) => (
              <>
                <li
                  className={`option ${
                    selectedLanguage.key === language.key ? "option-active" : ""
                  }`}
                  onClick={() => handleLanguageSelect(language)}
                  key={i}
                  value={language.key}
                >
                  {language.value}
                </li>
                <hr />
              </>
            ))}
          </ul>
        )}
      </div>
    </div>
  );
};

export default Languages;
