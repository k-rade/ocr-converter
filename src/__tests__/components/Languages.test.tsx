import { fireEvent, render } from "@testing-library/react";
import Languages from "../../components/Languages";

describe("Languages", (): void => {
  it("should render", (): void => {
    const { baseElement } = render(
      <Languages selectedLanguage="eng" setLanguage={jest.fn} />
    );

    expect(baseElement).toMatchSnapshot();
  });

  it("should call set language", (): void => {
    const mockSetLanguage = jest.fn();
    const { getByTestId } = render(
      <Languages selectedLanguage="eng" setLanguage={mockSetLanguage} />
    );

    fireEvent.change(getByTestId("languages"), { target: { value: "srp" } });

    expect(mockSetLanguage).toHaveBeenNthCalledWith(1, "srp");
  });
});
