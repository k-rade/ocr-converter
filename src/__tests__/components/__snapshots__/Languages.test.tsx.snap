// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Languages should render 1`] = `
<body>
  <div>
    <div
      class="flex justify-between flex-wrap gap-4 items-baseline"
    >
      <label
        class="text-white flex-[1.25] whitespace-nowrap"
        for="languages"
      >
        Character specific language
      </label>
      <select
        class="focus:ring-blue-500 border-0 text-md rounded focus:border-blue-500 outline-0 p-2 px-3 text-white bg-neutral-800 flex-1"
        data-testid="languages"
        id="languages"
      >
        <option
          value="afr"
        >
          Afrikaans
        </option>
        <option
          value="amh"
        >
          Amharic
        </option>
        <option
          value="ara"
        >
          Arabic
        </option>
        <option
          value="asm"
        >
          Assamese
        </option>
        <option
          value="aze"
        >
          Azerbaijani
        </option>
        <option
          value="aze_cyrl"
        >
          Azerbaijani - Cyrillic
        </option>
        <option
          value="bel"
        >
          Belarusian
        </option>
        <option
          value="ben"
        >
          Bengali
        </option>
        <option
          value="bod"
        >
          Tibetan
        </option>
        <option
          value="bos"
        >
          Bosnian
        </option>
        <option
          value="bul"
        >
          Bulgarian
        </option>
        <option
          value="cat"
        >
          Catalan; Valencian
        </option>
        <option
          value="ceb"
        >
          Cebuano
        </option>
        <option
          value="ces"
        >
          Czech
        </option>
        <option
          value="chi_sim"
        >
          Chinese - Simplified
        </option>
        <option
          value="chi_tra"
        >
          Chinese - Traditional
        </option>
        <option
          value="chr"
        >
          Cherokee
        </option>
        <option
          value="cym"
        >
          Welsh
        </option>
        <option
          value="dan"
        >
          Danish
        </option>
        <option
          value="deu"
        >
          German
        </option>
        <option
          value="dzo"
        >
          Dzongkha
        </option>
        <option
          value="ell"
        >
          Greek, Modern (1453-)
        </option>
        <option
          value="eng"
        >
          English
        </option>
        <option
          value="epo"
        >
          Esperanto
        </option>
        <option
          value="est"
        >
          Estonian
        </option>
        <option
          value="fin"
        >
          Finnish
        </option>
        <option
          value="fra"
        >
          French
        </option>
        <option
          value="gle"
        >
          Irish
        </option>
        <option
          value="hat"
        >
          Haitian; Haitian Creole
        </option>
        <option
          value="heb"
        >
          Hebrew
        </option>
        <option
          value="hin"
        >
          Hindi
        </option>
        <option
          value="hun"
        >
          Hungarian
        </option>
        <option
          value="ind"
        >
          Indonesian
        </option>
        <option
          value="isl"
        >
          Italian
        </option>
        <option
          value="jpn"
        >
          Japanese
        </option>
        <option
          value="kat"
        >
          Georgian
        </option>
        <option
          value="kor"
        >
          Korean
        </option>
        <option
          value="kur"
        >
          Kurdish
        </option>
        <option
          value="lao"
        >
          Lao
        </option>
        <option
          value="lat"
        >
          Latvian
        </option>
        <option
          value="lit"
        >
          Lithuanian
        </option>
        <option
          value="mal"
        >
          Malayalam
        </option>
        <option
          value="mkd"
        >
          Macedonian
        </option>
        <option
          value="mlt"
        >
          Maltese
        </option>
        <option
          value="mya"
        >
          Burmese
        </option>
        <option
          value="nep"
        >
          Nepali
        </option>
        <option
          value="nor"
        >
          Norwegian
        </option>
        <option
          value="pol"
        >
          Polish
        </option>
        <option
          value="por"
        >
          Portuguese
        </option>
        <option
          value="ron"
        >
          Romanian; Moldavian
        </option>
        <option
          value="rus"
        >
          Russian
        </option>
        <option
          value="san"
        >
          Sanskrit
        </option>
        <option
          value="slk"
        >
          Slovak
        </option>
        <option
          value="slv"
        >
          Slovenian
        </option>
        <option
          value="spa"
        >
          Spanish; Castilian
        </option>
        <option
          value="srp"
        >
          Serbian
        </option>
        <option
          value="srp_latn"
        >
          Serbian - Latin
        </option>
        <option
          value="swa"
        >
          Swahili
        </option>
        <option
          value="swe"
        >
          Swedish
        </option>
        <option
          value="syr"
        >
          Syriac
        </option>
        <option
          value="tur"
        >
          Turkish
        </option>
        <option
          value="tha"
        >
          Thai
        </option>
        <option
          value="ukr"
        >
          Ukrainian
        </option>
        <option
          value="uzb"
        >
          Uzbek
        </option>
        <option
          value="uzb_cyrl"
        >
          Uzbek - Cyrillic
        </option>
        <option
          value="vie"
        >
          Vietnamese
        </option>
        <option
          value="yid"
        >
          Yiddish
        </option>
      </select>
    </div>
  </div>
</body>
`;
